sys/uri_parser/uri_parser.c

```diff
@@ -54,6 +54,11 @@ static char *_consume_scheme(uri_parser_result_t *result, char *uri,
     result->scheme = uri;
     result->scheme_len = p - uri;
 
+    /* check if URI is long enough to contain a '://' */
+    if ((uri_end - p) <= 3) {
+        return NULL;
+    }
+
     /* check if authority part exists '://' */
     if ((p[1] != '\0') && (p[2] != '\0') && (p[1] == '/') && (p[2] == '/')) {
```

Comment:
I think you can remove the checks for '\0' here.  They are a left-over from a time, where the uri-parser was operating on a zero terminated buffer.

Reply:
Should be fixed, thanks.